tools/pihole_toml_to_markdown.py

```diff
@@ -158,37 +216,65 @@ def parse_toml_with_comments(filepath):
-            if "\n" in value:
-                indented_value = "\n".join("    " + v for v in value.splitlines()).strip()
+            if "\n" in example_display_value:
+                indented_value = "\n".join("    " + v for v in example_display_value.splitlines()).strip()
                 documentation.append(f"      {key} = {indented_value}")
             else:
-                documentation.append(f"      {key} = {value}")
+                documentation.append(f"      {key} = {example_display_value}")
             documentation.append("    ```")
 
             # CLI example tab
             documentation.append(f'=== "CLI"')
             documentation.append("    ```shell")
-            if "\n" in value and value.strip().startswith("["):
-                # Flatten multi-line array to single line for CLI
-                array_str = "".join(value.split())
-                documentation.append(f"    sudo pihole-FTL --config {full_key} '{array_str}'")
+            if example_display_value.strip().startswith("["):
+                # This is an array - need to escape quotes for CLI
+                if "\n" in example_display_value:
+                    # Flatten multi-line array to single line
+                    array_str = "".join(example_display_value.split())
+                else:
+                    array_str = example_display_value
+                # Escape the internal quotes for shell use
+                escaped_array = array_str.replace('"', '\\"')
+                documentation.append(f'    sudo pihole-FTL --config {full_key} "{escaped_array}"')
             else:
-                documentation.append(f"    sudo pihole-FTL --config {full_key} {value}")
+                documentation.append(f"    sudo pihole-FTL --config {full_key} {example_display_value}")
             documentation.append("    ```")
 
             # Environment variable example tab (for Docker Compose)
             documentation.append(f'=== "Environment (Docker Compose)"')
             documentation.append("    ```yaml")
             documentation.append("    environment:")
-            yaml_value = value.replace('"',"'")
-            if "\n" in yaml_value:
-                yaml_value = f"|\n        " + "\n      ".join(yaml_value.splitlines())
-            documentation.append(f"      {env_var}: {yaml_value}")
+            
+            # Convert TOML-style array to proper YAML format
+            if example_display_value.strip().startswith("["):
+                # This is an array - convert to YAML format with pipe syntax
+                # Extract content between [ and ]
+                array_content = example_display_value.strip()[1:-1].strip()
+                if array_content:
+                    # Format as YAML multiline string with pipe syntax
+                    # Use smart splitting that respects quoted strings
+                    items = parse_array_items(array_content)
+                    documentation.append(f"      {env_var}: |")
```

Comment:
```suggestion
                    documentation.append(f"      {env_var}: |-")
```

Shouldn't we use `|-`  for removing the final newline?

Reply:
Good question well asked. I didn't know about this option

Reply:
Both ways ultimately produce the same output in the toml file

```
  dnsmasq_lines = [
    "address=/example.com/",
    "address=/example.org/",
    "address=/example.net/"
  ] ### CHANGED (env), default = []
```

If you would prefer we didn't send the final new line then that's also fine and I can update my other docker docs PR to follow suit